Content.Server/Polymorph/Systems/PolymorphSystem.cs

```diff
@@ -177,8 +180,10 @@ private void OnPolymorphedTerminating(Entity<PolymorphedEntityComponent> ent, re
     /// <returns>The new entity, or null if the polymorph failed.</returns>
     public EntityUid? PolymorphEntity(EntityUid uid, PolymorphConfiguration configuration)
```

Comment:
This function has been re-organized, with some more internal documetnation. This is to more cleanly handle the entity-swapping proceedure into clear steps so we can reduce the potential surface of these weird polymorph bugs.